2014-04-22 07:41:53.395 ZMQ ./soa/service/zmq_endpoint.cc:447 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnecting to monitor/zeromq[0m
2014-04-22 07:41:53.400 ZMQ ./soa/service/zmq_endpoint.cc:494 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnected to tcp://ip-172-31-27-189.ec2.internal:24000[0m
2014-04-22 07:41:53.402 ZMQ ./soa/service/zmq_endpoint.cc:447 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnecting to monitor/zeromq[0m
2014-04-22 07:41:53.405 ZMQ ./soa/service/zmq_endpoint.cc:494 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnected to tcp://ip-172-31-27-189.ec2.internal:24000[0m
2014-04-22 07:41:53.411 ZMQ ./soa/service/zmq_endpoint.cc:447 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnecting to masterBanker/zeromq[0m
2014-04-22 07:41:53.417 ZMQ ./soa/service/zmq_endpoint.cc:494 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnected to tcp://ip-172-31-27-189.ec2.internal:22000[0m
augmentor frequency-cap-ex has connected
2014-04-22 07:41:53.460 ZMQ ./soa/service/zmq_endpoint.cc:447 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnecting to PostAuctionLoop/events[0m
2014-04-22 07:41:53.463 ZMQ ./soa/service/zmq_endpoint.cc:494 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnected to tcp://ip-172-31-27-189.ec2.internal:20000[0m
agent fixed-price-agent-ex_8335 connected to router
2014-04-22 07:41:53.467 ZMQ ./soa/service/zmq_endpoint.cc:447 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnecting to agentConfigurationService/listen[0m
2014-04-22 07:41:53.472 ZMQ ./soa/service/zmq_endpoint.cc:494 - bool Datacratic::ZmqNamedProxy::connect(const string&, Datacratic::ConnectionStyle) [1;31mconnected to tcp://ip-172-31-27-189.ec2.internal:19000[0m
got configuration message fixed-price-agent-ex_8335

--------------------------[Exception thrown]---------------------------
time:   2014-04-22-074153
type:   ML::Exception
pid:    8329; tid: 8422
what:   creatives require an ID to be specified
stack:
00: 0x0x7f07c440cdb7 at __cxa_throw + 0x27 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so + 0xdb7/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so(__cxa_throw+0x27) [0x7f07c440cdb7]
01: 0x0x7f07bfc18785 at RTBKIT::Creative::fromJson(Json::Value const&) + 0x215 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so + 0x51785/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so(_ZN6RTBKIT8Creative8fromJsonERKN4Json5ValueE+0x215) [0x7f07bfc18785]
02: 0x0x7f07bfc18f82 at RTBKIT::AgentConfig::createFromJson(Json::Value const&) + 0x792 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so + 0x51f82/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so(_ZN6RTBKIT11AgentConfig14createFromJsonERKN4Json5ValueE+0x792) [0x7f07bfc18f82]
03: 0x0x7f07bfc4cb68 at RTBKIT::AgentConfigurationListener::onMessage(std::vector<std::string, std::allocator<std::string> > const&) + 0x808 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so + 0x85b68/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so(_ZN6RTBKIT26AgentConfigurationListener9onMessageERKSt6vectorISsSaISsEE+0x808) [0x7f07bfc4cb68]
04: 0x0x7f07c3d37401 at Datacratic::ZmqNamedClientBusProxy::handleMessage(std::vector<std::string, std::allocator<std::string> > const&) + 0x71 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so + 0xa1401/home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so(_ZN10Datacratic22ZmqNamedClientBusProxy13handleMessageERKSt6vectorISsSaISsEE+0x71) [0x7f07c3d37401]
05: 0x0x7f07c24ca45b at Datacratic::ZmqEventSource::handleMessage(std::vector<std::string, std::allocator<std::string> > const&) + 0x6b in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xc645b/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic14ZmqEventSource13handleMessageERKSt6vectorISsSaISsEE+0x6b) [0x7f07c24ca45b]
06: 0x0x7f07c24caeb7 at Datacratic::ZmqEventSource::processOne() + 0x147 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xc6eb7/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic14ZmqEventSource10processOneEv+0x147) [0x7f07c24caeb7]
07: 0x0x7f07c24bdcca at Datacratic::MessageLoop::processOne() + 0xba in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xb9cca/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic11MessageLoop10processOneEv+0xba) [0x7f07c24bdcca]
08: 0x0x7f07c24bdcca at Datacratic::MessageLoop::processOne() + 0xba in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xb9cca/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic11MessageLoop10processOneEv+0xba) [0x7f07c24bdcca]
09: 0x0x7f07c24b9579 at Datacratic::MessageLoop::runWorkerThread() + 0xc9 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xb5579/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic11MessageLoop15runWorkerThreadEv+0xc9) [0x7f07c24b9579]
10: 0x0x7f07c24b9e58 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xb5e58/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(+0xb5e58) [0x7f07c24b9e58]
11: 0x0x7f07c298eb9d in /home/rtbkit/local/lib/libboost_thread.so.1.52.0 + 0x10b9d/home/rtbkit/local/lib/libboost_thread.so.1.52.0(+0x10b9d) [0x7f07c298eb9d]
12: 0x0x7f07c4616e9a in /lib/x86_64-linux-gnu/libpthread.so.0 + 0x7e9a/lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a) [0x7f07c4616e9a]
13: 0x0x7f07c2f463fd at clone + 0x6d in /lib/x86_64-linux-gnu/libc.so.6 + 0xf43fd/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7f07c2f463fd]


--------------------------[Exception thrown]---------------------------
time:   2014-04-22-074153
type:   ML::Exception
pid:    8329; tid: 8422
what:   parsing creative 3: creatives require an ID to be specified
stack:
00: 0x0x7f07c440cdb7 at __cxa_throw + 0x27 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so + 0xdb7/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so(__cxa_throw+0x27) [0x7f07c440cdb7]
01: 0x0x7f07bfc196b3 at RTBKIT::AgentConfig::createFromJson(Json::Value const&) + 0xec3 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so + 0x526b3/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so(_ZN6RTBKIT11AgentConfig14createFromJsonERKN4Json5ValueE+0xec3) [0x7f07bfc196b3]
02: 0x0x7f07bfc4cb68 at RTBKIT::AgentConfigurationListener::onMessage(std::vector<std::string, std::allocator<std::string> > const&) + 0x808 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so + 0x85b68/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libagent_configuration.so(_ZN6RTBKIT26AgentConfigurationListener9onMessageERKSt6vectorISsSaISsEE+0x808) [0x7f07bfc4cb68]

--------------------------[Exception thrown]---------------------------
time:   2014-04-22-074153
type:   ML::Exception
pid:    8329; tid: 8329
what:   couldn't bind to any port in range [12339,12340]
stack:
00: 0x0x7f07c440cdb7 at __cxa_throw + 0x27 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so + 0xdb7/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libexception_hook.so(__cxa_throw+0x27) [0x7f07c440cdb7]
01: 0x0x7f07c24a4092 at Datacratic::AcceptorT<Datacratic::SocketTransport>::listen(Datacratic::PortRange const&, std::string const&, Datacratic::PassiveEndpoint*, bool, int) + 0x1c2 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0xa0092/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic9AcceptorTINS_15SocketTransportEE6listenERKNS_9PortRangeERKSsPNS_15PassiveEndpointEbi+0x1c2) [0x7f07c24a4092]
02: 0x0x7f07c24a3e0b at Datacratic::PassiveEndpoint::init(Datacratic::PortRange const&, std::string const&, int, bool, bool, int) + 0x4b in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so + 0x9fe0b/home/rtbkit/rtbkit/build/x86_64/bin/../bin/libservices.so(_ZN10Datacratic15PassiveEndpoint4initERKNS_9PortRangeERKSsibbi+0x4b) [0x7f07c24a3e0b]
03: 0x0x7f07b1f6bd80 at RTBKIT::MockExchangeConnector::start() + 0x60 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libmock_exchange.so + 0x21d80/home/rtbkit/rtbkit/build/x86_64/bin/../bin/../bin/libmock_exchange.so(_ZN6RTBKIT21MockExchangeConnector5startEv+0x60) [0x7f07b1f6bd80]
04: 0x0x7f07c3d11dfd at RTBKIT::Router::startExchange(std::string const&, Json::Value const&) + 0x4d in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so + 0x7bdfd/home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so(_ZN6RTBKIT6Router13startExchangeERKSsRKN4Json5ValueE+0x4d) [0x7f07c3d11dfd]
05: 0x0x7f07c3d12280 at RTBKIT::Router::startExchange(Json::Value const&) + 0x40 in /home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so + 0x7c280/home/rtbkit/rtbkit/build/x86_64/bin/../bin/librtb_router.so(_ZN6RTBKIT6Router13startExchangeERKN4Json5ValueE+0x40) [0x7f07c3d12280]
06: 0x0x421908 at RTBKIT::RouterRunner::start() + 0xb8 in build/x86_64/bin/router_runner + 0x21908build/x86_64/bin/router_runner(_ZN6RTBKIT12RouterRunner5startEv+0xb8) [0x421908]
07: 0x0x4201ba at main + 0x7a in build/x86_64/bin/router_runner + 0x201babuild/x86_64/bin/router_runner(main+0x7a) [0x4201ba]
08: 0x0x7f07c2e7376d at __libc_start_main + 0xed in /lib/x86_64-linux-gnu/libc.so.6 + 0x2176d/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xed) [0x7f07c2e7376d]
09: 0x0x4208a9 in build/x86_64/bin/router_runner + 0x208a9build/x86_64/bin/router_runner() [0x4208a9]

terminate called after throwing an instance of 'ML::Exception'
  what():  couldn't bind to any port in range [12339,12340]
